src/auth/signout/SignOutPage.tsx

```diff
@@ -1,20 +1,23 @@
 import { useEffect } from 'react';
-import { SignOutState, userSignedOut } from 'src/auth/signout/sign-out';
+import { SignOutCause, SignOutRedirect, userSignedOut } from 'src/auth/signout/sign-out';
 import * as Nav from 'src/libs/nav';
 
 export const signOutCallbackLinkName = 'signout-callback';
 export const SignOutPage = () => {
   const { query } = Nav.useRoute();
   const { state } = query;
-  const decodedState: SignOutState | undefined = state ? JSON.parse(atob(state)).postLogoutRedirect : undefined;
+  const decoded: { signOutRedirect?: SignOutRedirect; signOutCause?: SignOutCause } = state
+    ? JSON.parse(atob(state))
+    : {};
+  const { signOutRedirect, signOutCause } = decoded;
```

Comment:
I'm wondering if there's a better way to handle this. It works and is typesafe, but its pretty ugly.

Reply:
I'm guessing these two lines could be combined?


```ts
const { signoutRedirect, signOutCause }: { signOutRedirect?: SignOutRedirect; signOutCause?: SignOutCause } = state
    ? JSON.parse(atob(state))
    : {};
```

Alternatively, pull some of this out into a hook:
```ts
interface SignOutState {
  signOutRedirect?: SignOutRedirect;
  signOutCause?: SignOutCause;
}

const useSignOutState(): SignOutState => {
  const { query } = Nav.useRoute();
  const { state } = query;
  return state ? JSON.parse(atob(state)) : {};
};
```